### PR TITLE
Using VM interval property to enable metrics

### PR DESCRIPTION
**Pull request by StrikingThirteen**
## Overview

Description:
This commit enables metrics collection and reporting based on the VM's metricsInterval. It also removes some of the obsolete code in MetricsUtils.

Why should this be merged: 
To enable collecting and reporting metrics.

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java [hunk 1]
```diff
@@ -31,43 +27,30 @@
     private static final MetricSet metricsJVMMem = new MemoryUsageGaugeSet();
     private static final MetricSet metricsJVMThread = new ThreadStatesGaugeSet();
 
-    private static Properties metricsProperties = new Properties();
+    private static int interval;
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: why is this static ?

Comment (StrikingThirteen) [reply runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: One of the properties that similar to metricsCollectionEnabled, and so on being used by other static methods of this util class. Mostly replaces: (private static Properties metricsProperties = new Properties();)

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java [hunk 2]
```diff
@@ -31,43 +26,37 @@
...
-            } catch (Exception e) {
-                log.error("Error processing METRICS_PROPERTIES {}: {}", propPath,
-                        e.toString());
-            }
+    private static void loadVmProperties() {
+        try {
+            interval = Integer.valueOf(System.getProperty("metricsInterval"));
+        } catch (NumberFormatException e) {
+            // interval defaults to -1 as disabled when the value cannot be
+            // extracted from VM property
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: log error

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java [hunk 3]
```diff
@@ -31,43 +26,37 @@
...
+            interval = -1;
+        }
+
+        if (interval > 0) {
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: what's the difference between metrics collection/reporting?

I think an application would want collection without reporting, or collection and reporting, but a +/- interval only enables/disables both options. Maybe use 0 for collection, but not reporting, neg-interval would be to disable everything and a pos-number would mean a reporting interval.

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java [hunk 4]
```diff
@@ -85,22 +74,19 @@ public static boolean isMetricsReportingSetUp(MetricRegistry metrics) {
...
      * Start metrics reporting via the Dropwizard 'CsvReporter' file writer.
      * Reporting can be turned on and off via the properties file described
-     * in loadPropertiesFile()'s docs.  The report interval and report
+     * in loadVmProperties()'s docs.  The report interval and report
      * directory cannot be altered at runtime.
      *
      * @param metrics Metrics registry
      */
     public static void metricsReportingSetup(MetricRegistry metrics) {
         metrics.counter(mpTrigger);
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: rename this, what is a mpTrigger ?

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java (hunk 2 above)
Comment (StrikingThirteen) [reply runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: Done.

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java (hunk 3 above)
Comment (StrikingThirteen) [reply runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: Done.
Collection without reporting might have limited value considering that dropwizard limited expressiveness at debugging at each point of time. However, it's a good idea to have that ability if application wants to have this ability for introducing a manual reporting. Added interval=0 for collection without reporting.

File: runtime/src/main/java/org/corfudb/util/MetricsUtils.java (hunk 4 above)
Comment (StrikingThirteen) [reply runtime/src/main/java/org/corfudb/util/MetricsUtils.java]: mpTrigger was intended by the author for filtering out some of the metrics. The logic requires investigating & clean-up yet avoiding side effects. That's a task that I will do outside the scope of this enabling metrics PR.

**Comment by Maithem** [comment]
@StrikingThirteen I see, please squash the history and lets get this in.

**Comment by StrikingThirteen** [comment]
@Maithem Squashed.

**Comment by Maithem** [comment]
@StrikingThirteen do we know why njection-framework  is failing ?

**Comment by StrikingThirteen** [comment]
@Maithem It was stale. New pre-checkin is in progress

**Pull request merged**